File: servers/dropwizard/src/main/java/com/slabstech/revive/server/dropwizard/services/GithubUserService.java
package com.slabstech.revive.server.dropwizard.services;

import com.slabstech.revive.server.dropwizard.core.GithubUser;

public interface GithubUserService {
    @GET("/users")
    public Call<List<GithubUser>> getUsers(
            @Query("per_page") int per_page,
            @Query("page") int page);

    @GET("/users/{username}")
    public Call<GithubUser> getUser(@Path("username") String username);
}

File: src/main/java/com/slabstech/revive/server/dropwizard/retrofit/guide/GithubUserService.kt
package com.slabstech.revive.server.dropwizard.retrofit.guide

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface GithubUserService {
    @GET("/githubusers")
    fun getGithubUsers(
        @Query("per_page") per_page: Int,
        @Query("page") page: Int
    ): Call<List<GithubUser?>?>?

    @GET("/githubusers/{username}")
    fun getGithubUser(@Path("username") username: String?): Call<GithubUser?>?
}